### PR TITLE
[SYCLomatic] Fix some migration bugs when using syclcompat headers (2)

### PR DESCRIPTION
**Pull request by zhiweij1**
Signed-off-by: Jiang, Zhiwei <zhiwei.jiang@intel.com>

### CODE REVIEW

File: clang/lib/DPCT/AnalysisInfo.cpp [hunk 1]
```diff
@@ -9,6 +9,7 @@
 #include "AnalysisInfo.h"
 #include "Diagnostics.h"
 #include "ExprAnalysis.h"
+#include "IncrementalMigrationUtility.h"
```
Comment (zhiweij1) [inline clang/lib/DPCT/AnalysisInfo.cpp]: remove

File: clang/lib/DPCT/Rewriters/Math/CallExprRewriterMath.h [hunk 2]
```diff
@@ -9,6 +9,7 @@
 #ifndef DPCT_REWRITERS_MATH_CALL_EXPR_REWRITER_MATH_H
 #define DPCT_REWRITERS_MATH_CALL_EXPR_REWRITER_MATH_H
 
+#include "AnalysisInfo.h"
```
Comment (zhiweij1) [inline clang/lib/DPCT/Rewriters/Math/CallExprRewriterMath.h]: remove

**Review by tomflinda — APPROVED** [review]
LGTM

File: clang/test/dpct/syclcompat_test2.cu [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+  cufftHandle plan1;
+  size_t* work_size;
+  int odist;
+  int ostride;
+  int * onembed;
+  int idist;
+  int istride;
+  int* inembed;
+  int * n;
+  // CHECK: plan1->commit(&*syclcompat::get_current_device().default_queue(), 3, n, inembed, istride, idist, onembed, ostride, odist, dpct::fft::fft_type::complex_double_to_real_double, 12, work_size);
```
Comment (zhimingwang36) [inline clang/test/dpct/syclcompat_test2.cu]: let's remove this &*

File: clang/lib/DPCT/Rewriters/Math/CallExprRewriterMath.h [hunk 4]
```diff
@@ -630,6 +630,18 @@ createMathAPIRewriterHost(
...
 }
 
+template <bool IsDouble> std::string getPiString() {
+  if constexpr (IsDouble) {
+    if (DpctGlobalInfo::useSYCLCompat())
+      return "(3.141592653589793115998)";
+    return "DPCT_PI";
+  } else {
+    if (DpctGlobalInfo::useSYCLCompat())
+      return "(3.14159274101257f)";
```
Comment (zhimingwang36) [inline clang/lib/DPCT/Rewriters/Math/CallExprRewriterMath.h]: can we use the M_PI in math.h?

**Comment by zhiweij1** [comment]
The newly added lit case passed on local test.
```
ksh-3.2$
ksh-3.2$ icpx -DBUILD_TEST -c -fsycl syclcompat_test2.dp.cpp
ksh-3.2$ echo $?
0
ksh-3.2$
```

**Pull request merged**